auslib/blobs/gmp.py

```diff
@@ -61,6 +61,15 @@ def getInnerXML(self, updateQuery, update_type, whitelistedDomains, specialForce
 
         return vendorXML
 
+    def validate(self, *args, **kwargs):
+        Blob.validate(self, *args, **kwargs)
+        hashes = dict(SHA1=40, SHA256=64, SHA512=128)
```

Comment:
Two things here:
1) I just found out this week that we expect to use sha384 in the near future. Please add this, too.
2) Since this will be shared between multiple blob types, it would be good to move the hash length definitions and actual checking of them to a separate function.